frontend_lib/src/helper.js

```diff
@@ -533,7 +533,7 @@ export const permissiveNumberEqual = (var1, var2) => {
 }
 
 export const createSpaceArborescence = spaceList => {
-  const spaceListWithChildren = spaceList.map(space => space.children ? space : { ...space, children: [] })
+  const spaceListWithChildren = spaceList.map(space => space.children ? { ...space } : { ...space, children: [] })
```

Comment:
```suggestion
  const spaceListWithChildren = spaceList.map(space => ({ ...space, children: space.children || [] }))
```

But I remain convinced that we should replace the children of spaces anyway. This function should not be used with spaces which have already children, or at least should not duplicate them.

Anyway, everything will work and is okay in practice here.

Reply:
`createSpaceTree`, I do not think that "Arborescence" is an english word.